src/lib/Server/Input/Parser/Criterion/Sibling.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace EzSystems\EzPlatformRest\Server\Input\Parser\Criterion;
+
+use eZ\Publish\API\Repository\LocationService;
+use eZ\Publish\API\Repository\Values\Content\Query\Criterion\Sibling as SiblingCriterion;
+use EzSystems\EzPlatformRest\Input\BaseParser;
+use EzSystems\EzPlatformRest\Input\ParsingDispatcher;
+use EzSystems\EzPlatformRest\Exceptions;
+
+class Sibling extends BaseParser
+{
+    /** @var \eZ\Publish\API\Repository\LocationService */
+    private $locationService;
+
+    public function __construct(LocationService $locationService)
+    {
+        $this->locationService = $locationService;
+    }
+
+    public function parse(array $data, ParsingDispatcher $parsingDispatcher): SiblingCriterion
+    {
+        if (!array_key_exists('SiblingsCriterion', $data)) {
+            throw new Exceptions\Parser('Invalid <SiblingCriterion> format');
+        }
```

Comment:
I'm not convinced it works right now the way it's described in the PR. See that you expect `Sibling` (according to your spec in the PR description) but check for Sibling*s*.
Correct form is singular `SiblingCriterion` IMHO, based on the actual criterion name.

Functional tests are in order IMHO. See my suggestion for @webhdx PR: https://github.com/webhdx/ezplatform-rest/compare/bf921e579751cce247561f54d812ed35f2e516bf...alongosz:tests-for-new-criteria
It's best that you coordinate effort (could be as a follow-up, but please fix the code here)

Reply:
> I'm not convinced it works right now the way it's described in the PR. See that you expect Sibling (according to your spec in the PR description) but check for Siblings.
Correct form is singular SiblingCriterion IMHO, based on the actual criterion name (or the other way around - adjust criterion name, but that is more work probably).

Fixed in https://github.com/ezsystems/ezplatform-rest/pull/40/commits/887083e00bcb4bc80fccd2cc2b73d2bc96d281c3

> Functional tests are in order IMHO. See my suggestion for @webhdx PR: webhdx/ezplatform-rest@bf921e5...alongosz:tests-for-new-criteria
It's best that you coordinate effort (could be as a follow-up, but please fix the code here)

The functional test will be added as a follow-up after https://github.com/ezsystems/ezplatform-rest/pull/39 merge